usersession/userd/launcher.go

```diff
@@ -123,6 +130,126 @@ func (s *Launcher) OpenURL(addr string, sender dbus.Sender) *dbus.Error {
 	return nil
 }
 
+// OpenDesktopEntryEnv implements the 'OpenDesktopEntryEnv' method of the 'io.snapcraft.Launcher'
+// DBus interface.
+func (s *Launcher) OpenDesktopEntryEnv(desktop_file_id string, env []string, sender dbus.Sender) *dbus.Error {
+	desktop_file, err := desktopFileIdToFilename(desktop_file_id)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	exec_command, err := readExecCommandFromDesktopFile(desktop_file)
+
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	args, err := shlex.Split(exec_command)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	// Passing exec variables between confined snaps raises unanswered questions and they are not required
+	// for the simple cases.  For now, we don't have support for passing them in the dbus API and drop
+	// them from the command.
+	// https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables
+	i := 0
+	for {
+		if strings.HasPrefix(args[i], "%") {
+			args = append(args[:i], args[i+1:]...)
+		} else {
+			i++
+		}
+
+		if i == len(args) {
+			break
+		}
+	}
+
```

Comment:
I had a look at our desktop files and they almost universally read:
```
Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/${SNAP_NAME}_${app_name}.desktop /snap/bin/${app_name} %U
```
Will those parse correctly? Should we add some sanity check that those parse exactly as we expect?